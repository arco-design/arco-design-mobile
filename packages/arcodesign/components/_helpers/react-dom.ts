import { ReactElement } from 'react';
import * as ReactDOM from 'react-dom';

function isObject(obj: any): obj is { [key: string]: any } {
    return Object.prototype.toString.call(obj) === '[object Object]';
}

export interface RootType {
    render: (element: ReactElement) => void;
    _unmount: () => void;
}
export interface RootTypeReact extends RootType {
    unmount?: () => void;
}
export type CreateRootFnType = (container: Element | DocumentFragment) => {
    render: (element: ReactElement) => void;
    unmount?: () => void;
};

// Cast ReactDOM to a version that might have createRoot and other properties
const typedReactDOM = ReactDOM as typeof ReactDOM & {
    createRoot?: CreateRootFnType;
    render?: (
        element: ReactElement,
        container: Element | DocumentFragment,
        callback?: () => void,
    ) => any;
    unmountComponentAtNode?: (container: Element | DocumentFragment) => boolean;
    __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED?: {
        usingClientEntryPoint?: boolean;
    };
};

let copyRender: (
    app: ReactElement,
    container: Element | DocumentFragment,
    createRootFunction?: CreateRootFnType,
) => RootType;

const updateUsingClientEntryPoint = (skipWarning?: boolean) => {
    // https://github.com/facebook/react/blob/17806594cc28284fe195f918e8d77de3516848ec/packages/react-dom/npm/client.js#L10
    // Avoid console warning
    const secretInternals = typedReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    if (isObject(secretInternals)) {
        secretInternals.usingClientEntryPoint = skipWarning;
    }
};

const createRootFn = typedReactDOM.createRoot;

const getRender =
    (createRootFunction: CreateRootFnType) =>
    (app: ReactElement, container: Element | DocumentFragment) => {
        updateUsingClientEntryPoint(true);
        const root = createRootFunction!(container);
        updateUsingClientEntryPoint(false);

        root.render(app);

        return {
            render: (elementToRender: ReactElement) => {
                root.render(elementToRender);
            },
            _unmount() {
                setTimeout(() => {
                    if (root && typeof root.unmount === 'function') {
                        root.unmount();
                    }
                });
            },
        };
    };

if (createRootFn) {
    // React 18
    copyRender = getRender(createRootFn);
} else if (
    typeof typedReactDOM.render === 'function' &&
    typeof typedReactDOM.unmountComponentAtNode === 'function'
) {
    // React 16/17
    copyRender = function (app: ReactElement, container: Element | DocumentFragment) {
        typedReactDOM.render!(app, container); // Use non-null assertion
        return {
            render: (elementToRender: ReactElement) => {
                typedReactDOM.render!(elementToRender, container); // Use non-null assertion
            },
            _unmount() {
                typedReactDOM.unmountComponentAtNode!(container); // Use non-null assertion
            },
        };
    };
} else {
    copyRender = (
        app: ReactElement,
        container: Element | DocumentFragment,
        createRootFunction?: CreateRootFnType,
    ) => {
        const defaultCb = () => {
            // Fallback if no rendering method is found
            console.error(
                'ArcoDesign: ReactDOM.createRoot() or ReactDOM.render() not found. ' +
                    'This usually means you are using an unsupported version of React, ' +
                    'or ReactDOM is not properly initialized. ' +
                    'ArcoDesign Mobile React requires React 16, 17, 18, or 19.',
            );
            return { render: (_element: ReactElement) => {}, _unmount: () => {} };
        };
        if (createRootFunction) {
            return getRender(createRootFunction)(app, container);
        }
        return defaultCb();
    };
}

export const render = copyRender;
