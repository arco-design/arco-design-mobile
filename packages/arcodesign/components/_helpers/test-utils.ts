import React, { useEffect, useRef } from 'react';
import { Promise as ES6Promise, Promise } from 'es6-promise';

/**
 * React 19 兼容的 act 包装器
 * 用于包装可能触发状态更新的异步操作
 */
export const actWrapper = (callback: () => void | ES6Promise<void>): ES6Promise<void> => {
    return new ES6Promise<void>((resolve, reject) => {
        try {
            // 尝试使用 react-dom/test-utils 的 act
            const testUtils = require('react-dom/test-utils');
            if (testUtils && typeof testUtils.act === 'function') {
                // 对于同步回调
                if (typeof callback === 'function') {
                    try {
                        const result = callback();
                        if (result && typeof result.then === 'function') {
                            testUtils
                                .act(() => result)
                                .then(() => resolve())
                                .catch(reject);
                        } else {
                            testUtils.act(() => {
                                callback();
                            });
                            resolve();
                        }
                    } catch (error) {
                        reject(error);
                    }
                } else {
                    resolve();
                }
            } else {
                const result = callback();
                if (result && typeof result.then === 'function') {
                    result.then(() => resolve()).catch(reject);
                } else {
                    resolve();
                }
            }
        } catch (error) {
            // 如果 act 不可用，直接执行回调
            try {
                const result = callback();
                if (result && typeof result.then === 'function') {
                    result.then(() => resolve()).catch(reject);
                } else {
                    resolve();
                }
            } catch (callbackError) {
                reject(callbackError);
            }
        }
    });
};

/**
 * 清理未处理的异步操作，防止内存泄漏警告
 */
export const cleanupAsync = (): ES6Promise<void> => {
    return new ES6Promise<void>(resolve => {
        setTimeout(() => resolve(), 0);
    });
};

/**
 * 安全的状态更新，检查组件是否已卸载
 */
export const createSafeStateUpdater = (isMountedRef: React.MutableRefObject<boolean>) => {
    return (updater: () => void) => {
        if (isMountedRef.current) {
            updater();
        }
    };
};

/**
 * 用于测试的 useEffect 清理函数
 */
export const useTestCleanup = (cleanupFn: () => void) => {
    const isMountedRef = useRef(true);

    useEffect(() => {
        return () => {
            isMountedRef.current = false;
            cleanupFn();
        };
    }, [cleanupFn]);

    return isMountedRef;
};
