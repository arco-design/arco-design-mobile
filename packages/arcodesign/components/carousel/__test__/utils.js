import { act } from 'react-dom/test-utils';
import { createMoveTouchEventObject, createStartTouchEventObject } from '../../../tests/helpers/mockEvent';

export function mockSwipeStart(map, clientXMap) {
    const { touchstart, touchmove } = clientXMap;
    act(() => {
        map.touchstart(createStartTouchEventObject({ x: touchstart, y: 0 }));
        map.touchmove(createMoveTouchEventObject({ x: touchmove, y: 0 }));
    });
}
export function mockSwipeEnd(wrapper, className, touchend) {
    wrapper.find(`.${className}`).simulate('touchEnd', createMoveTouchEventObject({ x: touchend, y: 0 }));
    act(() => {
        jest.advanceTimersByTime(600);
    });
    wrapper.update();
}

export function mockSwipe(map, wrapper, className, clientXMap) {
    const { touchend } = clientXMap;
    mockSwipeStart(map, clientXMap);
    mockSwipeEnd(wrapper, className, touchend);
}
