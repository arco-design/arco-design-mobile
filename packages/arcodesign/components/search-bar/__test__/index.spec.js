import React, { createRef } from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import { defaultContext } from '../../context-provider';
import SearchBar from '..';
import { delay } from '../../../tests/helpers/utils';

demoTest('search-bar');

mountTest(SearchBar, 'SearchBar');

const prefix = `${defaultContext.prefixCls}-search-bar`;
const actionBtnSelector = `.${prefix}-action-btn`;
const inputSelector = `.${prefix}-input`;
const clearIconSelector = `.${prefix}-clear`;
const searchIconSelector = `svg.${prefix}-search-icon`;
const associationContainerSelector = `.${prefix}-association-visible`;
const associationItemSelector = `.${prefix}-association-item`;
const highlightNodeSelector = `${associationItemSelector}-highlight`;

const useFakeTimersTest = (description, callback) => {
    it(description, () => {
        const root = document.createElement('div');
        document.body.appendChild(root);
        jest.useFakeTimers();

        callback(root);

        jest.useRealTimers();
        document.body.removeChild(root);
    });
};

// useInputLogic钩子中，onBlur和onFocus的回调都被延长了一个nextTick(20ms)，导致视图更新的延迟
// 因此需要手动延迟一段时间(比nextTick)长才能正确断言和显示隐藏相关的测试用例
const createInputDomSimulator = (inputDom, component) => {
    const originSimulate = inputDom.simulate;
    inputDom.simulate = (event, config) => {
        originSimulate.call(inputDom, event, config);
        delay(component, 25);
    };
    return inputDom;
};

describe('SearchBar', () => {
    useFakeTimersTest('component should render correctly', root => {
        const component = mount(<SearchBar />, { attachTo: root });
        // 输入框框被正常渲染
        expect(component.find(inputSelector).length).toBe(1);
        // 搜索icon被正常渲染
        expect(component.find(searchIconSelector).length).toBe(1);
        const inputDom = createInputDomSimulator(component.find('input'), component);
        inputDom.simulate('click');

        // 清除按钮被正常渲染
        expect(component.find(clearIconSelector).length).toBe(1);
        // 右侧actionBtn被正常渲染
        expect(component.find(actionBtnSelector).length).toBe(1);
    });

    useFakeTimersTest('actionBtn should render correctly', root => {
        const handleActionBtnClick = jest.fn();
        const component = mount(
            <SearchBar actionBtnText="搜索" onActionBtnClick={handleActionBtnClick} />,
            { attachTo: root },
        );

        const inputDom = createInputDomSimulator(component.find(inputSelector), component);

        // 测试actionBtnShowType为default的情况
        expect(component.find(actionBtnSelector).length).toBe(0);
        inputDom.simulate('focus');
        expect(component.find(actionBtnSelector).length).toBe(1);

        inputDom.simulate('blur');
        expect(component.find(actionBtnSelector).length).toBe(0);
        inputDom.simulate('change', { target: { value: '测试' } });
        expect(component.find(actionBtnSelector).length).toBe(1);
        inputDom.simulate('blur');
        expect(component.find(actionBtnSelector).length).toBe(1);

        // 测试actionBtnShowType为always的情况
        component.setProps({ actionBtnShowType: 'always' });
        inputDom.simulate('change', { target: { value: '' } });
        inputDom.simulate('blur');
        expect(component.find(actionBtnSelector).length).toBe(1);

        // 测试actionBtnShowType为focus的情况
        component.setProps({ actionBtnShowType: 'focus' });
        inputDom.simulate('blur');
        expect(component.find(actionBtnSelector).length).toBe(0);
        inputDom.simulate('focus');
        expect(component.find(actionBtnSelector).length);
        inputDom.simulate('blur');
        expect(component.find(actionBtnSelector).length).toBe(0);

        // 测试actionBtnShowType为value的情况
        component.setProps({ actionBtnShowType: 'value' });
        inputDom.simulate('change', { target: { value: '' } });
        expect(component.find(actionBtnSelector).length).toBe(0);
        inputDom.simulate('change', { target: { value: '测试' } });
        expect(component.find(actionBtnSelector).length).toBe(1);
        inputDom.simulate('blur');
        expect(component.find(actionBtnSelector).length).toBe(1);

        // 测试actionBtnShowType为none的情况
        component.setProps({ actionBtnShowType: 'none' });
        inputDom.simulate('focus');
        expect(component.find(actionBtnSelector).length).toBe(0);
        inputDom.simulate('change', { target: { value: '测' } });
        expect(component.find(actionBtnSelector).length).toBe(0);

        // 测试自定义append是否正常
        component.setProps({ actionBtnShowType: 'default' });
        const actionBtn = component.find(actionBtnSelector);
        inputDom.simulate('change', { target: { value: '测试' } });
        actionBtn.simulate('click');
        expect(handleActionBtnClick).toHaveBeenCalledWith('测试');

        component.setProps({
            append: <span id="test">测试</span>,
        });
        expect(component.find('#test').length).toBe(1);
    });

    it('association should render correctly', () => {
        const associationItems = [
            {
                content: '测试',
            },
        ];

        const component = mount(
            <SearchBar enableAssociation associationItems={associationItems} />,
        );

        expect(component.find(associationContainerSelector).length).toBe(0);

        // associationShowType为default
        const inputDom = component.find(inputSelector);
        inputDom.simulate('change', { target: { value: '测' } });
        expect(component.find(associationContainerSelector).length).toBe(1);

        inputDom.simulate('blur');
        expect(component.find(associationContainerSelector).length).toBe(0);

        inputDom.simulate('change', { target: { value: '' } });
        expect(component.find(associationContainerSelector).length).toBe(0);

        // associationShowType为value，blur也不会消失
        component.setProps({
            associationShowType: 'value',
        });
        inputDom.simulate('change', { target: { value: '测' } });
        expect(component.find(associationContainerSelector).length).toBe(1);
        inputDom.simulate('blur');
        expect(component.find(associationContainerSelector).length).toBe(1);
        inputDom.simulate('change', { target: { value: '' } });
        expect(component.find(associationContainerSelector).length).toBe(0);

        // associationShowType为focus, blur也消失
        component.setProps({
            associationShowType: 'focus',
        });
        inputDom.simulate('blur');
        expect(component.find(associationContainerSelector).length).toBe(0);
        inputDom.simulate('change', { target: { value: '测' } });
        expect(component.find(associationContainerSelector).length).toBe(1);

        component.setProps({
            associationShowType: 'always',
        });
        inputDom.simulate('blur');
        expect(component.find(associationContainerSelector).length).toBe(1);
        inputDom.simulate('change', { target: { value: '' } });
        expect(component.find(associationContainerSelector).length).toBe(1);
        inputDom.simulate('blur');
        expect(component.find(associationContainerSelector).length).toBe(1);
    });

    it('search-bar ref and on-event handler work correctly', () => {
        const associationItems = [
            {
                content: '测试',
            },
        ];

        const ref = createRef();
        const handleAssociationClick = jest.fn();
        const handleAssociationItemClick = jest.fn();

        const component = mount(
            <SearchBar
                enableAssociation
                ref={ref}
                associationItems={associationItems}
                onAssociationClick={handleAssociationClick}
                onAssociationItemClick={handleAssociationItemClick}
            />,
        );

        expect(component.find(associationContainerSelector).length).toBe(0);
        act(() => {
            ref.current.toggleAssociation();
        });
        component.update();
        expect(component.find(associationContainerSelector).length).toBe(1);
        act(() => {
            ref.current.toggleAssociation();
        });
        component.update();
        expect(component.find(associationContainerSelector).length).toBe(0);

        act(() => {
            ref.current.toggleAssociation(true);
        });
        component.update();
        expect(component.find(associationContainerSelector).length).toBe(1);

        component.find(associationContainerSelector).simulate('click');
        expect(handleAssociationClick).toBeCalled();
        const associationItemDoms = component.find(associationItemSelector);
        expect(associationItemDoms.length).toBe(1);

        const firstAssociationItem = associationItemDoms.at(0);
        firstAssociationItem.simulate('click');
        expect(handleAssociationItemClick).toHaveBeenCalledWith(associationItems[0], 0);
    });

    it('highlight prefix mode work correctly', () => {
        const associationItems = [
            {
                content: '测试',
            },
            {
                content: 'ArcoDesign',
            },
            {
                content: 'react',
            },
            {
                content: 'Ad',
            },
            {
                content: '一个测试组',
            },
        ];

        const component = mount(
            <SearchBar enableAssociation associationItems={associationItems} />,
        );

        expect(component.find(associationItemSelector).length).toBe(associationItems.length);
        expect(component.find(highlightNodeSelector).length).toBe(0);

        component.setProps({
            highlightMode: 'prefix',
        });
        const inputDom = component.find(inputSelector);

        inputDom.simulate('change', { target: { value: 'A' } });
        expect(component.find(highlightNodeSelector).length).toBe(2);
        inputDom.simulate('change', { target: { value: 'Arco' } });
        expect(component.find(highlightNodeSelector).length).toBe(2);
        inputDom.simulate('change', { target: { value: 'react' } });
        expect(component.find(highlightNodeSelector).length).toBe(1);

        inputDom.simulate('change', { target: { value: '' } });
        expect(component.find(highlightNodeSelector).length).toBe(0);

        component.setProps({
            highlightMode: 'contain',
        });
        inputDom.simulate('change', { target: { value: '测试' } });
        expect(component.find(highlightNodeSelector).length).toBe(2);

        component.setProps({
            highlightMode: (content, keyword, defaultClassName) =>
                content.includes(keyword) ? (
                    <span className={defaultClassName}>{content}</span>
                ) : (
                    content
                ),
        });
        component.update();
        expect(component.find(highlightNodeSelector).length).toBe(2);
    });

    it('highlight prefix mode work correctly', () => {
        const associationItems = [
            {
                content: '测试',
            },
            {
                content: 'ArcoDesign',
            },
            {
                content: 'react',
            },
            {
                content: 'great, refactor',
            },
            {
                content: '',
            },
        ];

        const component = mount(
            <SearchBar
                enableAssociation
                associationItems={associationItems}
                highlightMode="contain"
            />,
        );

        expect(component.find(associationItemSelector).length).toBe(associationItems.length);
        expect(component.find(highlightNodeSelector).length).toBe(0);
        const inputDom = component.find(inputSelector);
        inputDom.simulate('change', { target: { value: 'e' } });

        expect(component.find(highlightNodeSelector).length).toBe(4);
        inputDom.simulate('change', { target: { value: 're' } });
        expect(component.find(highlightNodeSelector).length).toBe(3);
        inputDom.simulate('change', { target: { value: '' } });
        expect(component.find(highlightNodeSelector).length).toBe(0);
    });

    it('render function should work correctly', () => {
        const renderSearchAssociation = Content => {
            return (
                <div>
                    <span id="test">Test</span>
                    {Content}
                </div>
            );
        };

        const renderSearchAssociationItem = (item, index, node) => {
            return (
                <>
                    <span id="test-item">Test</span>
                    {node}
                </>
            );
        };

        const associationItems = [
            {
                content: '测试',
            },
        ];

        const component = mount(
            <SearchBar
                enableAssociation
                associationItems={associationItems}
                renderAssociation={renderSearchAssociation}
                renderAssociationItem={renderSearchAssociationItem}
                highlightMode="unknown"
            />,
        );

        const inputDom = component.find(inputSelector);
        inputDom.simulate('change', { target: { value: 'test' } });

        expect(component.find('#test').length).toBe(1);
        expect(component.find('#test-item').length).toBe(1);
    });
});
