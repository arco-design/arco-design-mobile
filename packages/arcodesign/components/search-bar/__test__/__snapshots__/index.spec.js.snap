// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-bar demo test search-bar demo: action-btn.md renders correctly 1`] = `
<div>
  <div
    class="arco-radio-group"
    style="padding:16px;background-color:white"
  >
    <div
      class="arco-radio shape-circle inline"
    >
      <div
        class="radio-icon checked"
      >
        <svg
          class="arco-icon arco-icon-circle-checked "
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10 1c-5 0-9 4-9 9s4 9 9 9 9-4 9-9-4-9-9-9zm4.9 6.3L10 13.2s0 .1-.1.1l-.6.7c-.1.1-.2.1-.3.2-.1 0-.3 0-.4-.1l-.6-.7-.1-.1-2.8-2.4c-.2-.2-.2-.5-.1-.7l.6-.7c.2-.2.5-.2.7-.1l2.5 2.1L13.5 6c.2-.2.5-.2.7-.1l.7.6c.1.3.2.6 0 .8z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="radio-text"
      >
        default
      </div>
    </div>
    <div
      class="arco-radio shape-circle inline"
    >
      <div
        class="radio-icon"
      >
        <svg
          class="arco-icon arco-icon-circle-unchecked "
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 19c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9zm0-17c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="radio-text"
      >
        always
      </div>
    </div>
    <div
      class="arco-radio shape-circle inline"
    >
      <div
        class="radio-icon"
      >
        <svg
          class="arco-icon arco-icon-circle-unchecked "
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 19c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9zm0-17c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="radio-text"
      >
        none
      </div>
    </div>
    <div
      class="arco-radio shape-circle inline"
    >
      <div
        class="radio-icon"
      >
        <svg
          class="arco-icon arco-icon-circle-unchecked "
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 19c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9zm0-17c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="radio-text"
      >
        value
      </div>
    </div>
    <div
      class="arco-radio shape-circle inline"
    >
      <div
        class="radio-icon"
      >
        <svg
          class="arco-icon arco-icon-circle-unchecked "
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 19c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9zm0-17c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="radio-text"
      >
        focus
      </div>
    </div>
  </div>
  <div
    class="arco-search-bar-container all-border-box arco-search-bar-square"
  >
    <div
      class="arco-search-bar-wrap search border-half pc prefix"
    >
      <div
        class="arco-search-bar-prefix"
      >
        <svg
          class="arco-icon arco-icon-search arco-search-bar-search-icon"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M469.333 106.667C669.632 106.667 832 269.035 832 469.333c0 75.712-23.19 146.006-62.87 204.16l144.683 144.683a21.333 21.333 0 010 30.165l-36.202 36.203a21.333 21.333 0 01-30.166 0L706.56 743.68A361.259 361.259 0 01469.333 832c-200.298 0-362.666-162.368-362.666-362.667s162.368-362.666 362.666-362.666zm0 89.6c-150.826 0-273.066 122.24-273.066 273.066S318.507 742.4 469.333 742.4 742.4 620.16 742.4 469.333 620.16 196.267 469.333 196.267z"
            fill="currentColor"
          />
        </svg>
      </div>
      <input
        class="arco-search-bar-input"
        placeholder="请输入要查询的内容"
        type="search"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`search-bar demo test search-bar demo: association.md renders correctly 1`] = `
<div
  class="arco-search-bar-container all-border-box arco-search-bar-square"
>
  <div
    class="arco-search-bar-wrap search border-half pc prefix"
  >
    <div
      class="arco-search-bar-prefix"
    >
      <svg
        class="arco-icon arco-icon-search arco-search-bar-search-icon"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M469.333 106.667C669.632 106.667 832 269.035 832 469.333c0 75.712-23.19 146.006-62.87 204.16l144.683 144.683a21.333 21.333 0 010 30.165l-36.202 36.203a21.333 21.333 0 01-30.166 0L706.56 743.68A361.259 361.259 0 01469.333 832c-200.298 0-362.666-162.368-362.666-362.667s162.368-362.666 362.666-362.666zm0 89.6c-150.826 0-273.066 122.24-273.066 273.066S318.507 742.4 469.333 742.4 742.4 620.16 742.4 469.333 620.16 196.267 469.333 196.267z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      class="arco-search-bar-input"
      placeholder="请输入要查询的内容"
      type="search"
      value=""
    />
  </div>
  <div
    class="arco-search-bar-association"
  />
</div>
`;

exports[`search-bar demo test search-bar demo: custom-icon.md renders correctly 1`] = `
<div>
  <div
    class="arco-search-bar-container all-border-box arco-search-bar-square"
  >
    <div
      class="arco-search-bar-wrap search border-half pc"
    >
      <input
        class="arco-search-bar-input"
        placeholder="请输入要查询的内容"
        type="search"
        value=""
      />
    </div>
  </div>
  <div
    class="arco-search-bar-container all-border-box arco-search-bar-square"
    style="margin-top:20px"
  >
    <div
      class="arco-search-bar-wrap search border-half pc prefix"
    >
      <div
        class="arco-search-bar-prefix"
      >
        <svg
          class="arco-icon arco-icon-setting "
          fill="currentColor"
          height="1em"
          style="color:#86909c"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M512 46.55c34.517 0 68.565 3.797 101.696 11.263l19.136 4.288L692.48 198.08l146.837-16.043 13.27 14.443a465.493 465.493 0 01101.802 176.939l5.76 18.602-11.456 15.723L872.747 512l87.402 119.979-5.76 18.602A465.493 465.493 0 01852.587 827.52l-13.27 14.443L692.48 825.92l-59.648 135.979-19.136 4.288c-33.13 7.466-67.2 11.264-101.696 11.264-34.517 0-68.587-3.798-101.717-11.264l-19.136-4.31-7.894-17.941L331.52 825.92l-146.837 16.043-13.27-14.443A465.493 465.493 0 0169.611 650.56l-5.76-18.603 11.456-15.722L151.253 512 63.851 392.043l5.76-18.603a465.493 465.493 0 01101.802-176.96l13.27-14.443L331.52 198.08l59.627-135.957 19.136-4.31c33.13-7.466 67.2-11.264 101.717-11.264zm0 77.567a387.58 387.58 0 00-65.792 5.611L398.421 238.72a63.488 63.488 0 01-65.066 37.61l-117.483-12.863a387.947 387.947 0 00-66.048 114.837l70.165 96.32c16.214 22.272 16.214 52.48 0 74.752l-70.186 96.32a387.947 387.947 0 0066.069 114.837l117.504-12.864a63.488 63.488 0 0165.045 37.632l47.787 108.971a387.393 387.393 0 0065.792 5.61 387.33 387.33 0 0065.77-5.61l47.81-108.992a63.488 63.488 0 0165.066-37.61l117.483 12.863a387.947 387.947 0 0066.048-114.816l-70.165-96.341a63.488 63.488 0 010-74.752l70.165-96.341a387.947 387.947 0 00-66.048-114.816L690.624 276.33a63.488 63.488 0 01-65.045-37.611l-47.787-108.992a386.325 386.325 0 00-65.792-5.61zm0 193.942c106.752 0 193.195 86.869 193.195 193.941S618.752 705.92 512 705.92 318.805 619.093 318.805 512 405.248 318.08 512 318.08zm0 77.568c-63.808 0-115.627 52.053-115.627 116.373 0 64.299 51.819 116.373 115.627 116.373S627.627 576.32 627.627 512c0-64.299-51.819-116.373-115.627-116.373z"
          />
        </svg>
      </div>
      <input
        class="arco-search-bar-input"
        placeholder="请输入要查询的内容"
        type="search"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`search-bar demo test search-bar demo: custom-pend.md renders correctly 1`] = `
<div
  class="arco-search-bar-container all-border-box arco-search-bar-square"
>
  <div
    style="display:flex;align-items:center;font-size:14px;margin-right:10px"
  >
    City 
    <svg
      class="arco-icon arco-icon-arrow-down "
      height="1em"
      style="margin-left:5px;font-size:10px"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 676.864l346.944-346.965a21.333 21.333 0 0130.187 0l30.165 30.186a21.333 21.333 0 010 30.166L542.166 767.36a42.667 42.667 0 01-60.331 0L104.704 390.25a21.333 21.333 0 010-30.165l30.165-30.186a21.333 21.333 0 0130.187 0L512 676.864z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="arco-search-bar-wrap search border-half pc prefix"
  >
    <div
      class="arco-search-bar-prefix"
    >
      <svg
        class="arco-icon arco-icon-search arco-search-bar-search-icon"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M469.333 106.667C669.632 106.667 832 269.035 832 469.333c0 75.712-23.19 146.006-62.87 204.16l144.683 144.683a21.333 21.333 0 010 30.165l-36.202 36.203a21.333 21.333 0 01-30.166 0L706.56 743.68A361.259 361.259 0 01469.333 832c-200.298 0-362.666-162.368-362.666-362.667s162.368-362.666 362.666-362.666zm0 89.6c-150.826 0-273.066 122.24-273.066 273.066S318.507 742.4 469.333 742.4 742.4 620.16 742.4 469.333 620.16 196.267 469.333 196.267z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      class="arco-search-bar-input"
      placeholder="请输入要查询的内容"
      type="search"
      value=""
    />
  </div>
  <div
    class="arco-avatar-wrapper ultra-small circle"
    style="margin-left:13px"
  >
    <div
      class="arco-avatar ultra-small circle default-overlap"
    >
      <svg
        class="arco-icon arco-icon-user-fill arco-avatar-default"
        fill="currentColor"
        height="1em"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.5 10.833c2.301 0 5 1.786 5 5v2.5c0 .46-.373.834-.833.834H3.333a.833.833 0 01-.833-.834v-2.5c0-3.211 2.699-5 5-5h5zM10 1.25a4.167 4.167 0 110 8.333 4.167 4.167 0 010-8.333z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`search-bar demo test search-bar demo: index.md renders correctly 1`] = `
<div>
  <div
    class="arco-radio-group"
    style="padding:16px;background-color:white"
  >
    <div
      class="arco-radio shape-circle inline"
    >
      <div
        class="radio-icon checked"
      >
        <svg
          class="arco-icon arco-icon-circle-checked "
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10 1c-5 0-9 4-9 9s4 9 9 9 9-4 9-9-4-9-9-9zm4.9 6.3L10 13.2s0 .1-.1.1l-.6.7c-.1.1-.2.1-.3.2-.1 0-.3 0-.4-.1l-.6-.7-.1-.1-2.8-2.4c-.2-.2-.2-.5-.1-.7l.6-.7c.2-.2.5-.2.7-.1l2.5 2.1L13.5 6c.2-.2.5-.2.7-.1l.7.6c.1.3.2.6 0 .8z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="radio-text"
      >
        square
      </div>
    </div>
    <div
      class="arco-radio shape-circle inline"
    >
      <div
        class="radio-icon"
      >
        <svg
          class="arco-icon arco-icon-circle-unchecked "
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 19c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9zm0-17c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="radio-text"
      >
        round
      </div>
    </div>
  </div>
  <div
    class="arco-search-bar-container all-border-box arco-search-bar-square"
  >
    <div
      class="arco-search-bar-wrap search border-half pc prefix"
    >
      <div
        class="arco-search-bar-prefix"
      >
        <svg
          class="arco-icon arco-icon-search arco-search-bar-search-icon"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M469.333 106.667C669.632 106.667 832 269.035 832 469.333c0 75.712-23.19 146.006-62.87 204.16l144.683 144.683a21.333 21.333 0 010 30.165l-36.202 36.203a21.333 21.333 0 01-30.166 0L706.56 743.68A361.259 361.259 0 01469.333 832c-200.298 0-362.666-162.368-362.666-362.667s162.368-362.666 362.666-362.666zm0 89.6c-150.826 0-273.066 122.24-273.066 273.066S318.507 742.4 469.333 742.4 742.4 620.16 742.4 469.333 620.16 196.267 469.333 196.267z"
            fill="currentColor"
          />
        </svg>
      </div>
      <input
        class="arco-search-bar-input"
        placeholder="请输入要查询的内容"
        type="search"
        value=""
      />
    </div>
  </div>
</div>
`;
